docs/defaulting.md

```diff
@@ -0,0 +1,249 @@
+# Ahead of Time libvirt Defaulting on the Cluster/API Level
+
+This Proposal fills a gap between KubeVirt as Runtime and Higher level
+management, like proposed in
+[docs/config-and-instances.md](docs/config-and-instances.md). It lays out a
+design which overcomes the necessity for asynchronous specification updates,
+which has it's origin in the fact that a lot of the defaulting is done on the
+host by libvirt, so after scheduling.
+
+With the proposal here, which will provide a synchronous way of fully
+populating a VM specification, we come much closer to the meaning of a Pod
+specification in Kubernetes.
+
+Note that a fully populated VM specification in the context of this proposal, is
+a specification which has filled out all fields, which influence how the
+emulated machine looks like for the Guest OS. For instance libvirt network
+configurations may differ on a host to host basis, e.g. because on one host a
+path to an Ethernet device is `/dev/eth0` and on another one `/dev/eth1`,
+although in both cases we connect to the same network. Such backend
+configuration specific fields are not of interest for this proposal.
+
+The goals of this proposal are:
+
+ * Synchronous defaulting of all VM specification related fields, which
+   influence how a Guest OS sees a VM.
+ * As a consequence, enforcing the meaning of the VM specification as
+   specification. We then don't have to change the specification after it was
+   provided, or force users to extract additional data from the status field,
+   which they would have to put into the VM specification in subsequent runs,
+   to get exactly the same VM.
+ * Give us more control over what we want to expose in `status`, instead of
```

Comment:
The `.status` field is independent of this defaulting approach.
I do see your point how defaulting could have an impact on what the `.status` field exposes. But to me that order of reasoning is the wrong way round, because defaulting (especially ahead of time) is a higher-level concept than reflecting a domains status.

We also need to recognize that some flows will not use an ahead-of-time defaulting approach, because it's not needed/suited everywhere.
Thus I'd suggest to keep the `.status` field discussion out of this document.